tools/offline_ota_image_builder/README.md

```diff
@@ -0,0 +1,173 @@
+
+Option `--output <OUTPUT>` specify where to save the exported tar archive of built image.
+
+User must at least specifies one of `--write-to` and `--output`, or specify them together.
+
+#### Usage 1: Build offline OTA image and export it as tar archive
+
+```bash
+# current folder layout: venv ota-client
+(venv) $ cd ota-client
+(venv) $ sudo -E env PATH=$PATH python3 -m tools.offline_ota_image_builder --image=p1:p1_image.tgz:ver_20230808 --image=p2:p2_image.tgz:ver_20230808 --output=t2.tar
+```
+
+This will build the offline OTA image with `p1_image.tgz` and `p2_image.tgz`, which are for ECU `p1` and `p2`, and export the built image as `t2.tar` tar archive.
+
+### Usage 2: Build the offline OTA image and create external cache source dev
+
+```bash
+# current folder layout: venv ota-client
+(venv) $ cd ota-client
+(venv) $ sudo -E env PATH=$PATH python3 -m tools.offline_ota_image_builder --image=p1:p1_image.tgz:ver_20230808 --image=p2:p2_image.tgz:ver_20230808 --write-to=/dev/<target_dev>
+```
+
+This will build the offline OTA image with `p1_image.tgz` and `p2_image.tgz`, which are for ECU `p1` and `p2`, and then prepare the `/dev/<target_dev>` as external cache source device(ext4 filesystem labelled with `ota_cache_src`) with image rootfs exported to the filesystem on `/dev/<target_device>`.
+
+### Usage 3: Build the offline OTA image, export it as tar archive and prepare external cache source dev
+
+```bash
+# current folder layout: venv ota-client
+(venv) $ cd ota-client
+(venv) $ sudo -E env PATH=$PATH python3 -m tools.offline_ota_image_builder --image=p1:p1_image.tgz:ver_20230808 --image=p2:p2_image.tgz:ver_20230808 --output=t2.tar --write-to=/dev/<target_dev>
+```
+
+## Manual
+
+```text
+usage: offline_ota_image_builder [-h] --image <ECU_NAME>:<IMAGE_PATH>[:<IMAGE_VERSION>]
+                                 [-o <OUTPUT_PATH>] [-w <DEVICE>] [--confirm-write-to]
+
+Helper script that builds offline OTA image with given OTA image(s) as external cache source or for
+offline OTA use.
+
+options:
+  -h, --help            show this help message and exit
+  --image <ECU_NAME>:<IMAGE_PATH>[:<IMAGE_VERSION>]
+                        OTA image for <ECU_ID> as tar archive(compressed or uncompressed), this
+                        option can be used multiple times to include multiple images.
+  -o <OUTPUT_PATH>, --output <OUTPUT_PATH>
+                        save the generated image rootfs into tar archive to <OUTPUT_PATH>.
+  -w <DEVICE>, --write-to <DEVICE>
+                        write the image to <DEVICE> and prepare the device as external cache source
+                        device.
+  --confirm-write-to    writing generated to <DEVICE> without inter-active confirmation, only valid
+                        when used with -w option.
+```
```

Comment:
It is better to move this block right after `### Build image and export` section to understand the tool usage, and it is better to add an explanation after that.